### DISCUSSION TITLE
Possible? - Plugin Automation clarification needed

### DISCUSSION THREAD

**Discussion by Fookes74** [maxileith/homebridge-appletv-enhanced#272] [Q&A] [answered]
Firstly, what a fantastic plugin. Love it. 

I'm trying to set up an automation where, if I'm playing a plex movie via the plex app on my TV, I can dim lights when I play something and then restore brightness if I pause or stop the movie. At the moment, it looks like a can do an automation such as this but when anything is playing from any app on the TV - which I don't want. I only want this for the Plex app. Is it do-able or will the plugin setup not allow me to do this?

**Comment by maxileith** [comment]
It is possible. You can query the activity ID of the Apple TV in a shortcut automation and then just check whether or not the activity ID represents Plex or not.

**Reply by Fookes74** [reply]
Thanks for the response. Much appreciated. You may have guessed the follow-up question though:

I have zero idea how to set that up (I’ve never used shortcuts app). Any chance of some guidance?

**Reply by kali8183** [reply]
Hi, i also would like to to that, but dont understand about "query the activity id" ?

**Reply by Fookes74** [reply]
Ditto. I don’t understand the instructions I was given.

**Reply by maxileith** [reply]
This example automation is executed whenever the play status is being activated. The section where the comment is, is only executed when Disney+ is the currently open app ... so this is the part where you want to put your Disney specific play automation logic. You could add a section for other apps as well.

![image](https://github.com/maxileith/homebridge-appletv-enhanced/assets/44269066/5523e75a-bf72-4975-9c18-d87402d45301)

![image](https://github.com/maxileith/homebridge-appletv-enhanced/assets/44269066/d3057fb6-5a93-464b-9dbb-1cb99a245a3c)

For Plex, just use the respective activity ID. You can easily figure that out by playing a movie with Plex and querying the activity ID.

![image](https://github.com/maxileith/homebridge-appletv-enhanced/assets/44269066/a37fc368-f437-4299-b0e4-24e47309be29)

**Comment by Fookes74** [comment]
Thanks, but I’m still not understanding this fully. The first link shows an initial part of a home automation. How are you completing this automation after this page?The next two links are from a shortcut automation I guess, but it’d be appreciated if you could show fully how this is set up from start to finish. How are these two automations linked? Chris Dafforne

**Reply by maxileith** [reply]
Please do not create a new comment when replying to another comment. Instead use the reply function.

Never reply via E-Mail! This just makes a big mess in GitHub … always.

**Reply by maxileith** [reply]
Besides that, the example is complete. The first screenshot is the trigger and the second one the automation itself.

**Reply by Fookes74** [reply]
Ok, duly noted on the replies via GitHub. I still need clarification. 
Is the first image showing an automation created in the home app? If so, please confirm what the full automation is. It shows (as far as I can see): 
if Motion detected = play…. Then what?

The second and third images seem to be shortcut screenshots. What is the full shortcut? I’m not understanding how this is created. Thanks.

**Reply by maxileith** [reply]
> if Motion detected = play…. Then what? 

The second screenshot is the "Then what?" part. I think you are missing the following button:

![image](https://github.com/maxileith/homebridge-appletv-enhanced/assets/44269066/ce9ffa3b-7a90-446d-ab9c-86c02d660e15)

The third screenshot is the shortcuts app. This is just a simple way of identifying the activity ID of the currently opened app. This is how you can figure out the activity ID of Plex.

**Reply by kali8183** [reply]
I succeeded, thank you very much, great work!

**Reply by Fookes74** [reply]
Could you possibly assist me. I still have little clue how to set this up. Not a user of shortcuts so would be grateful of a walkthrough. Just not understanding what I’m meant to be doing and where!

**Reply by maxileith** [reply]
I have already provided a walkthrough. Where are you stuck?

The automation itself is setup in Home.

The shortcut to find out the Plex activity ID is setup in the shortcuts app.

**Reply by Fookes74** [reply]
With great respect, you’ve posted a couple of screenshots with some info and assumed I can fill in the gaps and that I have the same level of understanding of the app(s) you’re using that you do. 

I don’t - I have zero experience of the shortcuts app.

I’ll post some screenshots up to where I get stuck. I’m still unclear if I’m using just the shortcuts app or using the home app also. I’ll start with what I think you’ve done in the shortcuts app. Images and text to follow….

**Reply by Fookes74** [reply]
Here’s where I got to. I started it in home app, converted to shortcut. I have no idea what number I’m meant to put in here though. 
![IMG_2305](https://github.com/maxileith/homebridge-appletv-enhanced/assets/37369430/c09e7141-9499-44b3-9e49-7cefa0f84ef7)

**Reply by maxileith** [reply]
I have explained here how to get the activity ID of Plex (the screenshot is from the shortcuts app ... as mentioned before).

> For Plex, just use the respective activity ID. You can easily figure that out by playing a movie with Plex and querying the activity ID.

> ![image](https://github.com/maxileith/homebridge-appletv-enhanced/assets/44269066/a37fc368-f437-4299-b0e4-24e47309be29)

---

> With great respect, you’ve posted a couple of screenshots with some info and assumed I can fill in the gaps and that I have the same level of understanding of the app(s) you’re using that you do.

I think it is fair to assume some basic knowledge here. I am not the Apple support and not here to explain the bare minimum. I am here to assist with plugin specific questions, not the shortcut app or home app itself.

**Reply by Fookes74** [reply]
That’s fine. I did ask the other user who had sorted it to assist me - hence reply to him. You answered it. If you don’t want to help then I’m unsure why you replied! 
I’ll see if the other user is willing to assist.

**Reply by kali8183** [reply]
Hi, to get the activity ID of Plex you need to make another shortcut directly in the shortcut app, not home app.
Then you need to launch a plex movie and click the "play" button at the bottom right on your shortcut page, and you'll get the ID that you need to put in the home app shortcut.
Hope I helped you.

**Reply by Fookes74** [reply]
Hi, @kali8183 - thanks for the response. You mention ANOTHER shortcut. What’s the first shortcut?

Here’s where I am, right from the start.  I’ve created this in the shortcuts app. If I press play at this point nothing happens. I’m running Plex on my client whilst I’m pressing play in this shortcut. I have t saved the shortcut btw, I’m just pressing play on this screen. 
![IMG_2310](https://github.com/maxileith/homebridge-appletv-enhanced/assets/37369430/d8e68beb-9b23-439d-a5c0-b39a35a3a505)

**Comment by kali8183** [comment, marked as answer]
The first shortcut is the one you already created in the home app. But it missing the plex activity ID.
Its why you need to create a second one in shortcut app just to get the ID.

**Reply by Fookes74** [reply]
Think I’ve got it. Thanks for the clarity on that. as a quick aside, these instructions are to set lights or whatever when Plex starts playing. I take it a further automation (using the same Plex activity ID) to revert lights or whatever back to their original state would work (but using ‘stop’ or ‘pause’ as the trigger? Or would you need a different Plex Activity ID?

Appreciate your help on this. Much clearer.

**Reply by kali8183** [reply]
Yes the ID is unique so you can make others shortcuts with same ID when pause, stop, or something else.

**Reply by Fookes74** [reply]
That’s brilliant. Thanks so much. Greatly appreciated.
